### ISSUE TITLE
Kundera mongoDB native query support

### ISSUE THREAD

**Issue by sreekanth0041** [Impetus/kundera#683]
I am trying to get the data using native support but the error is

```
Exception in thread "main" com.impetus.kundera.query.QueryHandlerException: java.lang.UnsupportedOperationException: Native query support is not enabled in mongoDB
    at com.impetus.client.mongodb.query.MongoDBQuery.populateEntities(MongoDBQuery.java:172)
    at com.impetus.kundera.query.QueryImpl.fetch(QueryImpl.java:986)
    at com.impetus.kundera.query.QueryImpl.getResultList(QueryImpl.java:163)
    at com.test.Test2.main(Test2.java:80)
```

My code is

```
Query query = em.createNativeQuery("db.resturant.find({contact: {\"$elemMatch\" :  {\"Mobile\" : \"9965652365\"}}})");
query.getResultList();
```

Please help me.

**Comment by devender-yadav** [comment]
For executing Native Queries in MongoDB using Kundera you need to write scripts. Please refer this test case 

https://github.com/impetus-opensource/Kundera/blob/273c13342ddd1aceed0cd23504649926ce8fdb84/src/kundera-mongo/src/test/java/com/impetus/client/crud/PersonMongoTest.java#L155

Please let me know if you face any further issue.

**Comment by sreekanth0041** [comment]
Thank you for giving the reply devender.
I tried with above but problem is not resolved

``````
EntityManagerFactory emf = Persistence
                .createEntityManagerFactory("mango");
EntityManager em = emf.createEntityManager();
 Map<String, Client<Query>> clients = (Map<String, Client<Query>>) em.getDelegate();
        Client client = clients.get("mango");

        String jScript = "db.system.js.save({ _id: \"echoFunction\",value : function(x) { return x; }})";
        Object result = ((MongoDBClient)client).executeScript(jScript);

        String findOneJScript = "db.resturant.findOne()";
        result = ((MongoDBClient) client).executeScript(findOneJScript);
        String findAllJScript = "db.resturant.find( { \"state\": \"Telangana\" } )";
        result = ((MongoDBClient)client).executeScript(findAllJScript);

        try
        {
            em.createNativeQuery(findAllJScript, Restaurant.class).getResultList();
        }
        catch (Exception e)
        {
           e.printStackTrace();
        }

        em.close();
    emf.close();
    ```
Error is
```com.impetus.kundera.query.QueryHandlerException: java.lang.UnsupportedOperationException: Native query support is not enabled in mongoDB
    at com.impetus.client.mongodb.query.MongoDBQuery.populateEntities(MongoDBQuery.java:172)
    at com.impetus.kundera.query.QueryImpl.fetch(QueryImpl.java:986)
    at com.impetus.kundera.query.QueryImpl.getResultList(QueryImpl.java:163)
    at com.test.Test2.main(Test2.java:121)
``````

**Comment by devender-yadav** [comment]
Hi @sreekanth0041  no need to call  
em.createNativeQuery(findAllJScript, Restaurant.class).getResultList(); 

U will get your result in 
 result = ((MongoDBClient)client).executeScript(findAllJScript);

You can not use em.createNativeQuery() method as exception itself says "Native query support is not enabled in mongoDB". Please go through the test case again.

**Comment by sreekanth0041** [comment]
I am getting the result data like below

```
{ "_mongo" : { "slaveOk" : false , "host" : "EMBEDDED" , "_writeMode" : "commands" , "_hasWriteCommands" : true} , "_db" : { "_mongo" : { "slaveOk" : false , "host" : "EMBEDDED" , "_writeMode" : "commands" , "_hasWriteCommands" : true} , "_name" : "laalsa"} , "_collection" : { "_mongo" : { "slaveOk" : false , "host" : "EMBEDDED" , "_writeMode" : "commands" , "_hasWriteCommands" : true} , "_db" : { "_mongo" : { "slaveOk" : false , "host" : "EMBEDDED" , "_writeMode" : "commands" , "_hasWriteCommands" : true} , "_name" : "laalsa"} , "_shortName" : "resturant" , "_fullName" : "laalsa.resturant"} , "_ns" : "laalsa.resturant" , "_query" : { } , "_fields" :  null  , "_limit" : 0.0 , "_skip" : 0.0 , "_batchSize" : 0.0 , "_options" : 0.0 , "_cursor" :  null  , "_numReturned" : 0.0 , "_special" : false}
```

Now how to get the collection from the above.

**Comment by mevivs** [comment]
@sreekanth0041 
Based on parsing above json result(using http://json.parser.online.fr/), i think you looking for 

```
"_shortName":"resturant",
"_fullName":"laalsa.resturant"
```

Correct?   You need to use, jackson or google's json parser to parse and extract relevant json column!

HTH,
-Vivek

**Comment by chhavigangwal** [comment]
@sreekanth0041 

You can also modify your script to retrieve collection objects in following manner:

```
String findAllJScript = "db.resturant.find( { \"state\": \"Telangana\" } ).toArray()";
```

and then parse the json object.

Hope that Helps!
Chhavi

**Comment by chhavigangwal** [comment]
Fixed. Released with Kundera-2.15.

Chhavi
